eng/scripts/initialize.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# This script is used to execute mypy within a tox environment. Depending on which package is being executed against,
+# a failure may be suppressed.
+
+import os
+from pathlib import Path
+from helpers import call, str2bool
+import argparse
+import shutil
+
+root_dir = Path.resolve(Path(__file__) / "../../..")
+
+
+def copy(src: Path, dest: Path) -> None:
+    dest.parent.mkdir(parents=True, exist_ok=True)
+    shutil.copy(src, dest)
+
+
+def main(update_to_latest_typespec: bool, build_artifacts_path: Path) -> None:
+    # Install global dependencies
+    call("npm install -g pnpm@8.3.1")
+    call("npm install -g autorest@3.6.3")
+    call("npm install -g @typespec/compiler@0.49.0")
```

Comment:
I feel we shouldn't pin versions here, it's hard to update when new releases come out. Could we just have the `package.json` handle the compiler version? We might have to `npx` run the compiler, but I think that would work

Reply:
We should avoid global installs and rely on package.json and appropriate lock/shrinkwrap files.

Reply:
Could we move the versioning into a lock / shrink file here then? I am just pretty confident that this won't get updated every time we update our dependencies